File: BackToBasics/Recursion/multiplication.cpp
/*
Multiplication (Recursive)
##Given two integers M & N, calculate and return their multiplication using recursion.
You can use subtraction and addition for your calculation. No other operators are allowed.
Input Format:
Line 1: Integer M
Line 2: Integer N
Output Format:
M * N
Contstraints:
0 <= M <= 1000
0 <= N <= 1000
Sample Input 1:
3
5
Sample Output 1:
12
Sample Input 2:
4
0
sample Output 2:
0
*/
#include <iostream>
using namespace std;
int multiplication(int a, int b)
{
    if (a == 0 || b == 0)
    {
        return 0;
    }
    if (a < b)
    {
        return b + multiplication(a - 1, b);
    }
    return a + multiplication(a, b - 1);
}
int main()
{
    // Write your code here
    int a, b;
    cin >> a >> b;
    int ans = multiplication(a, b);
    cout << ans;
    return 0;
}

File: DSA/Recursion/Multiplication.java
import java.util.Scanner;

public class Multiplication {
	public static void main(String[] args) {
		Scanner s = new Scanner(System.in);
		int m = s.nextInt();
		int n = s.nextInt();
		System.out.println(solution.multiplyTwoIntegers(m, n));
	}
}


public class solution {

	public static int multiplyTwoIntegers(int m, int n){
		// Write your code here
        if(m==0 || n==0){ // if any of them is 0 then ans is 0
            return 0;
        }
        if(m == 1){
            return n;
        }
        if(n == 1){
            return m;
        }
        if(m<n){ // n + add n (m-1) times
            return n + multiplyTwoIntegers(m-1, n);
        }else{   // m + add m (n-1) times
            return m + multiplyTwoIntegers(m, n-1);
        }
	
	}
}

